src/runtime.ts

```diff
@@ -93,6 +93,31 @@ function mapInternal<A extends ValueType, T extends ValueType>(
   return value;
 }
 
+export function filterDeep<A extends ValueType, T extends ValueType>(
+  value: A,
+  predicate: (a: any) => a is T
+): readonly A[] {
+  return filterInternal(value, predicate);
+}
+
+function filterInternal<A extends ValueType, T extends ValueType>(
+  value: A,
+  predicate: (a: any) => a is T
+): readonly A[] {
+  if (predicate(value)) {
```

Comment:
this will not find recursive structures

Reply:
does it matter though? Isn't the idea just to see if `value is T`?

Reply:
I'm not sure I follow. Am I mistaken that if you have a parent and and child both for which `predicate` is true this will only return the parent node and not *all* nodes where `predicate` is true

Reply:
oh, good point. So that's why `map` and `pmap` keep iterating even if there's a match. I'll do the change.